### DISCUSSION TITLE
launch or terminate a call programatically

### DISCUSSION THREAD

**Discussion by solucionesuno** [InnovateAsterisk/Browser-Phone#407] [General]
Hi,

is any way to lunch a call, or teminate a call with some action?

**Comment by InnovateAsterisk** [comment]
There are various public functions in the global space and event hooks, called "web_hooks". 
The main function for setting up a call is ```DialByLine(type, buddy, numToDial, CallerID, extraHeaders)``` documented here:
https://github.com/InnovateAsterisk/Browser-Phone/blob/12061962af0ba09899c432cdc1e18a2f2297740e/Phone/phone.js#L8710

The web hooks are all listed in the sample Index.html here:
https://github.com/InnovateAsterisk/Browser-Phone/blob/master/Phone/index.html
Within this file, there is a script that demonstrates all the available event hooks, and also a provisioning sample using the ```phoneOptions``` object in the global space.
